# Generated by Django 5.1.1 on 2024-12-11 16:01

import contextlib
import arango.exceptions
from django.db import migrations

import typing
from django.conf import settings
from arango.client import ArangoClient

if typing.TYPE_CHECKING:
    from .. import settings




def remove_old_view(*args, **kw):
    client = ArangoClient(settings.ARANGODB_HOST_URL, request_timeout=600)
    try:
        db = client.db(
            settings.ARANGODB_DATABASE + "_database",
            settings.ARANGODB_USERNAME,
            settings.ARANGODB_PASSWORD,
            verify=True,
        )
        collection = db.collection('nvd_cve_vertex_collection')
        collection.delete_index('cve_search_inv', ignore_missing=True)
    except arango.exceptions.ServerConnectionError as e:
        assert e.http_code == 404 and e.error_code == 1228, f"unable to access arangodb database: {e.message}"
    except arango.exceptions.IndexDeleteError as e:
        assert e.http_code == 404 and e.error_code == 1203, f"deletion fail: {e.message}"

class Migration(migrations.Migration):

    initial = True

    dependencies = [
        ('server', '0002_remove_vulmatch_view'),
    ]

    operations = [
        migrations.RunPython(
            code=remove_old_view,
            reverse_code=migrations.RunPython.noop,
        ),
    ]
